### PR TITLE
enhancement: Use Hook for Paymaster Bundler Contract

### PR DESCRIPTION
**Pull request by arsood**
**What changed? Why?**

- Use generateContractHook for Paymaster Bundler

**How has it been tested?**

- Locally

### CODE REVIEW

File: template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { baseSepolia } from 'viem/chains';
+import { generateContractHook } from '@/hooks/contracts';
+import { entryPoint } from '../constants';
+import { PaymasterBundlerABI } from './PaymasterBundlerABI';
+
+/**
+ * Returns contract data for the PaymasterBundler contract.
+ */
+export const usePaymasterBundlerContract = generateContractHook({
+  abi: PaymasterBundlerABI,
```
Comment (arsood) [inline template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts:10]: @0xAlec there is a type error with the ABI. Can you investigate?

File: template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  [baseSepolia.id]: {
+    chain: baseSepolia,
+    address: entryPoint,
```
Comment (arsood) [inline template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts:13]: @0xAlec is this the correct constant to use here?

Comment (0xAlec) [reply template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts]: updated

File: template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts (hunk 1 above)
Comment (0xAlec) [reply template/web/app/paymaster-bundler/_contracts/usePaymasterBundlerContract.ts:10]: fixed

**Comment by 0xAlec** [comment]
can't edit the PR description but also added a guide to the paymaster experience

**Pull request merged**